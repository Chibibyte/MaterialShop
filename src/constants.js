export default {
    BUTTONICONS: {
        AUTHENTICATION: 'mdi-login-variant',
        ADDRESS: 'mdi-package-variant',
        BANKACCOUNT: 'mdi-bank',
        SHIPPING: 'mdi-truck',
        PAYMENT: 'mdi-cash',
        SUMMARY: 'mdi-text-box-check-outline',
        CHECKOUT_COMPLETE: 'mdi-check',
        ADD: 'mdi-plus',
        REMOVE: 'mdi-minus',
        CANCEL: 'mdi-window-close',
        CONFIRM: 'mdi-check',
        SHOPPING: 'mdi-shopping',
        SUCCESS: 'mdi-check',
        FAILURE: 'mdi-window-close',
        WARNING: 'mdi-alert-circle-outline'
    }
}