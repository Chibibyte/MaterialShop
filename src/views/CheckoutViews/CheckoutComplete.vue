<template>
  <CheckoutBase
    :buttonFunction="() => $store.commit('checkout/clearCheckoutStates')"
    :next_disabled="false"
    title="Thank you!"
    subtitle="You order will be processed immediately."
    next_name="Shopping"
    next_target="/"
    :next_icon="$constants.BUTTONICONS.SHOPPING"
    :icon="$constants.BUTTONICONS.CHECKOUT_COMPLETE"
  ></CheckoutBase>
</template>

<script>
import CheckoutBase from "@/views/CheckoutViews/CheckoutBase.vue";

export default {
  name: "CheckoutComplete",
  components: {
    CheckoutBase,
  },
};
</script>