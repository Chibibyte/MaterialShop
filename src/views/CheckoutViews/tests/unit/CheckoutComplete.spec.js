
import Vuex from 'vuex'
import { shallowMount, createLocalVue } from '@vue/test-utils';
import CheckoutComplete from '@/views/CheckoutViews/CheckoutComplete.vue';
import constants from '@/constants.js'

const localVue = createLocalVue();
localVue.use(Vuex);
localVue.prototype.$constants = constants;

let wrapper;

beforeEach(() => {
    wrapper = shallowMount(CheckoutComplete, { localVue });
})

// initial
describe('Component', () => {
    test('is instantiated', () => {
        expect(wrapper.isVueInstance).toBeTruthy();
    });

    test('default snapshot', () => {
        expect(wrapper.element).toMatchSnapshot();
    })
});



