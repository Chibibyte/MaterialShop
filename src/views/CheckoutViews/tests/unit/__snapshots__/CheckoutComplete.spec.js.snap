// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component default snapshot 1`] = `
<checkoutbase-stub
  icon="mdi-check"
  next_icon="mdi-shopping"
  next_name="Shopping"
  next_target="/"
  subtitle="You order will be processed immediately."
  title="Thank you!"
/>
`;
